compiler/src/iree/compiler/InputConversion/StableHLO/StableHLOToLinalgExt.cpp

```diff
@@ -439,6 +440,167 @@ struct ReverseOpConversion final
+    }
+
+    auto inputs = op.getInputs();
+    if (inputs.size() != 1) {
+      return rewriter.notifyMatchFailure(op, "more than one input");
+    }
+
+    auto input0 = inputs.front();
+    auto input0Ty = input0.getType().cast<ShapedType>();
+    auto init0 = op.getInitValues().front();
+    auto init0Ty = init0.getType().cast<ShapedType>();
+
+    auto window = extract1DVector(op.getWindowDimensions());
+    llvm::SmallVector<int64_t, 4> reduceAxes;
+    for (int i = 0, s = window.size(); i < s; ++i) {
+      if (window[i] == 1)
+        continue;
+      if (window[i] == input0Ty.getDimSize(i)) {
+        reduceAxes.push_back(i);
+        continue;
+      }
+
+      // Arguably it's still beneficial across a partial window, but this
+      // depends on performance characteristics.
+      return rewriter.notifyMatchFailure(op, "not length-1 or full width");
+    }
+
+    if (reduceAxes.size() != 1) {
+      return rewriter.notifyMatchFailure(op, "non singular reduction axis");
+    }
+
+    const int64_t reduceAxis = reduceAxes.front();
+
+    if (!op.getPadding()) {
+      return rewriter.notifyMatchFailure(op, "no padding values found");
+    }
+
+    auto padding = extract1DVector(*op.getPadding());
+    if (padding.size() < reduceAxis * 2) {
+      return rewriter.notifyMatchFailure(op, "no padding along reduction");
+    }
+
+    for (int i = 0, s = padding.size(); i < s; i += 2) {
+      if (i == reduceAxis * 2)
+        continue;
+      if (padding[i] != 0 || padding[i + 1] != 0) {
+        return rewriter.notifyMatchFailure(op,
+                                           "padding along non-reduction axis");
+      }
+    }
+
+    bool isPrefix =
+        padding[reduceAxis * 2] == (input0Ty.getDimSize(reduceAxis) - 1);
+    bool isPostfix =
+        padding[reduceAxis * 2 + 1] == (input0Ty.getDimSize(reduceAxis) - 1);
+
+    if (isPrefix == isPostfix) {
+      return rewriter.notifyMatchFailure(op, "is not purely prefix or postfix");
+    }
+
+    llvm::SmallVector<Value> outputs;
+    llvm::SmallVector<Value> outputDynDims;
+    for (int i = 0; i < input0Ty.getRank(); ++i) {
+      if (input0Ty.isDynamic(i)) {
+        outputDynDims.push_back(
+            rewriter.createOrFold<tensor::DimOp>(op.getLoc(), input0, i));
+      }
+    }
+
+    llvm::SmallVector<Value> init;
+    llvm::SmallVector<int64_t> initDims;
+    llvm::SmallVector<Value> initDynDims;
+    for (int i = 0; i < input0Ty.getRank(); ++i) {
+      if (i == reduceAxis)
+        continue;
+      initDims.push_back(input0Ty.getDimSize(i));
+      if (initDims.back() == ShapedType::kDynamic) {
+        initDynDims.push_back(
+            rewriter.createOrFold<tensor::DimOp>(op.getLoc(), input0, i));
+      }
+    }
+
+    outputs.push_back(rewriter.create<tensor::EmptyOp>(
+        op.getLoc(), input0Ty.getShape(), input0Ty.getElementType(),
+        outputDynDims));
+
+    // TODO(suderman): Change this to something better
```

Comment:
This seems a bit vague

Reply:
Yeah, I think this was a carry over as well. Deleted.